src/kubernetes_cluster/proof/cluster_safety.rs

```diff
@@ -40,7 +40,10 @@ pub open spec fn etcd_object_is_well_formed(key: ObjectRef) -> StatePred<Self> {
             else if key.kind == RoleView::kind() { RoleView::unmarshal(s.resources()[key]).is_Ok() }
             else if key.kind == SecretView::kind() { SecretView::unmarshal(s.resources()[key]).is_Ok() }
             else if key.kind == ServiceView::kind() { ServiceView::unmarshal(s.resources()[key]).is_Ok() }
-            else if key.kind == StatefulSetView::kind() { StatefulSetView::unmarshal(s.resources()[key]).is_Ok() }
+            else if key.kind == StatefulSetView::kind() { 
+                &&& StatefulSetView::unmarshal(s.resources()[key]).is_Ok() 
+                &&& StatefulSetView::unmarshal(s.resources()[key]).get_Ok_0().state_validation()
```

Comment:
It is a bit surprising that we need to say the state_validation holds for stateful set here, but do not say it holds for other types like role binding or role. Can you comment on why?

Reply:
No longer needed and I will delete it